src/dev/src/app/utils/demo.component.ts

```diff
@@ -0,0 +1,206 @@
+        <pre *ngIf="!resultHidden"><code [highlight]="resultsData() | json"></code></pre>
+      </div>
+      <div class="col-md-6 col-lg-6 col-xl-6 col-sm-12">
+        <p>Code (<button class="btn btn-link p-0" (click)="toggleCode()">{{ codeHidden ? 'View Code' : 'Hide Code'}}</button>)</p>
+        <pre *ngIf="!codeHidden"><code [highlight]="codeBlock"></code></pre>
+      </div>
+    </div>`,
+})
+export class DemoComponent implements AfterViewInit, AfterContentInit {
+  @Input() componentData: DemoComponentData;
+  @Input() disabled: boolean = false;
+  @Input() codeHidden: boolean = false;
+  @Input() resultHidden: boolean = false;
+  @ViewChild('demo', { read: ViewContainerRef })
+  _vcr;
+
+  title: string;
+  sourceCode: string;
+  models: object;
+  params: any;
+  canDisable: boolean;
+  codeBlock: string;
+
+  constructor(private _compiler: Compiler) {}
+
+  ngAfterContentInit() {
+    this.title = this.componentData.title;
+    this.sourceCode = this.componentData.source;
+    this.models = this.componentData.models;
+    this.params = this.componentData.params;
+    this.canDisable = this.componentData.canDisable;
+
+    if (!this.params.disabled) {
+      this.params['disabled'] = this.disabled;
+    }
+
+    let codeBlocks = this.extractCodeBlocks(this.sourceCode);
+    this.codeBlock = jsBeautify.html(codeBlocks.length > 0 ? codeBlocks.join('') : this.sourceCode);
+  }
+
+  ngAfterViewInit() {
+    const _params = this.params;
+    const _models = this.models;
+
+    const tmpCmp = Component({ template: this.sourceCode })(
+      class DemoSubComponent {
+        public params: object = _params;
+        public models: object = _models;
+      }
+    );
+
+    const tmpModule = NgModule({
+      imports: [BrowserAnimationsModule, CommonModule, FormsModule, FeruiModule],
+      declarations: [tmpCmp],
+    })(class {});
+
+    this._compiler.compileModuleAndAllComponentsAsync(tmpModule).then(factories => {
+      const f = factories.componentFactories[factories.componentFactories.length - 1];
+      this._vcr.createComponent(f);
+    });
+  }
+
+  resultsData() {
+    const data: object = {};
+    if (this.canDisable) {
+      data['params'] = { disabled: this.params.disabled };
+    }
+    data['models'] = {};
+    for (let modelName in this.models) {
+      data['models'][modelName] = this.models[modelName];
+    }
+    return data;
+  }
+
+  concatResultModels(models): Array<any> {
+    const results: Array<any> = [];
+    for (let modelName in models) {
+      results.push({
+        'field-name': modelName,
+        value: models[modelName],
+      });
+    }
+    return results;
+  }
+
+  toggleCode() {
+    this.codeHidden = !this.codeHidden;
+  }
+
+  toggleResult() {
+    this.resultHidden = !this.resultHidden;
+  }
+
+  disable(disabled: boolean) {
+    if (this.canDisable) {
+      this.params.disabled = disabled;
+    }
+  }
+
+  private extractCodeBlocks(code: string) {
```

Comment:
Maybe explain what does the two private functions there.

Reply:
Fixed

Reply:
Fixes